Wordsmith project description (#12778)
apache/datafusion: README.md

DataFusion is an extensible query engine written in [Rust] that
uses [Apache Arrow] as its in-memory format.

The DataFusion project provides libraries and binaries for data system
developers and end users such as data engineers and analysts. This repository
contains libraries for system developers. The following related projects work

> out of the box and target end users.

Reviewer: The DataFrame API that this project ships, for whom it is? (mentioned 13 lines below)

Also, "The target audience for the DataFusion crates in this repository are developers" (9 lines below) sounds like a repetition now.

Reviewer 2: > The DataFrame API that this project ships, for whom it is? (mentioned 13 lines below)

It is a good question -- I think @timsaucer and @Omega359 uses the Rust DataFrame library as is.

Reviewer 3: I do use the dataframe api almost exclusively. I say almost because I do allow sql expressions in my library which are converted to Expr's in DF and used via the dataframe api.

Reviewer 4: I do use the DataFrame API exclusively, but I work almost entirely via `datafusion-python`.

DataFusion is an extensible query engine written in [Rust] that
uses [Apache Arrow] as its in-memory format.

> The DataFusion project provides a suite of libraries and binaries to meet the needs of data system developers and end
> users such as data engineers and analysts. There are libraries in this repository for those looking for a custom experience
> as well as the below prepackaged projects which work out of the box and target end users.

Reviewer 2: I spent some time wordsmithing / condensing. What do you think of this version?

```suggestion
The DataFusion project provides libraries and binaries for data system
developers and end users such as data engineers and analysts. This repository
contains libraries for system developers. The following related projects work
out of the box and target end users.
```

- [DataFusion Comet](https://github.com/apache/datafusion-comet/) is an accelerator for Apache Spark based on
  DataFusion.

"Out of the box,"

> DataFusion offers [SQL] and [`Dataframe`] APIs, excellent [performance],

Reviewer: nit: Will this render with a linebreak after `"Out of the box,"`?

Reviewer 3: I don't believe it will. Markdown is a bit ambiguous on this and different impl's handle this differently however most either will force a newline in rendered view if the line ends with two spaces (now that is a horrible pattern) or a back slash. This comment editor is an example of something that does put text on a newline 
when a line break is encountered.
